src/stores/index.js

```diff
@@ -796,15 +793,9 @@ const Store = Reflux.createStore({
    * @param {Object} connection - A connection.
    */
   _saveConnection(connection) {
-    connection.replicaSet = connection.replicaSet || '';
-    connection.mongodbDatabaseName = connection.mongodbDatabaseName || '';
-
     this.state.currentConnection = connection;
     this.state.fetchedConnections.add(connection);
-    this.state.connections[connection._id] = connection.getAttributes({
-      props: true,
-      derived: true
-    });
+    this.state.connections[connection._id] = { ...connection._values };
```

Comment:
so the `connection._values` is going to have the correct `.replicaSet` and have it set to be an empty string if none?

Reply:
@lrlna yes. `getAttributes` method returns only properties that have some value but `_values` attribute stores all properties including those that have `undefined` value. Therefore when we save a favorite we save all properties, and when we switch between favorites we update all current connection properties. In the case of `undefined` value, we unset a property.

Reply:
ahhhh! gotcha! this is way better than doing it manually then!